src/libsystemd/sd-journal/journal-file.c

```diff
@@ -1781,7 +2147,17 @@ uint64_t journal_file_entry_array_n_items(Object *o) {
         if (sz < offsetof(Object, entry_array.items))
                 return 0;
 
-        return (sz - offsetof(Object, entry_array.items)) / sizeof(uint64_t);
+        return JOURNAL_HEADER_COMPACT(f->header) ?
+                (sz - offsetof(Object, entry_array.compact)) / sizeof(uint32_t) :
+                (sz - offsetof(Object, entry_array.items)) / sizeof(uint64_t);
```

Comment:
should use `sizeof(entry_array.compact[0])` and `sizeof(entry_array.items[0])`

Reply:
This would be a bigger change I think. There's nothing in the coding style about this and the rest of the file is using this style so I prefer to stick to that.